### PR TITLE
fix(grainfmt): Maintain comments in function arguments

### LINKED ISSUE #983: Grainfmt: Comment in between arguments on own lines gets moved to weird place
Given the code:

```grain
Result.expect(
    "Unexpected error when writing to stdout",
    // Is stdout correct here?
    File.fdWrite(File.stdout, "Status: 500\n\nInternal Server Error")
)
```

Grainfmt currently generates:

```grain
Result.expect(
    "Unexpected error when writing to stdout",
    File.fdWrite(File.stdout, "Status: 500\n\nInternal Server Error"),
) // Is stdout correct here?
```

But I expect the comment to remain in place because it is on a different line than the parameter.

### PR DESCRIPTION
**Pull request by marcusroberts**
Fixes #983 

Use the new iterator to format function arguments in function application

### CODE REVIEW

File: compiler/src/formatting/format.re [hunk 1]
```diff
@@ -23,7 +24,7 @@ type iterator_item_type =
   | IteratedRecordData
   | IteratedValueBindings;
 type expression_parent_type =
-  | ConditionalExpression
+  | ExpressionInParens
```
Comment (ospencer) [inline compiler/src/formatting/format.re:27]: How about `GroupedExpression`?

**Review by ospencer — COMMENTED** [review]
Looks good, just one comment!

File: compiler/src/formatting/format.re (hunk 1 above)
Comment (marcusroberts) [reply compiler/src/formatting/format.re]: I like it.  Done.

**Review by ospencer — APPROVED** [review]
Thanks @marcusroberts!

**Review by phated — APPROVED** [review]
LGTM! Thanks

**Pull request merged**